### PR TITLE
fix: sets data marketing to false in case of non-selection

### PR DESCRIPTION
**Pull request by jonybur**
## **Description**

Avoids showing the experience enhancer modal after onboarding.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2704

## **Manual testing steps**

1. Onboard
2. User should not see experience enhancer after onboarding

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by salimtb — DISMISSED** [review]

**Review by sahar-fehri — DISMISSED** [review]

**Comment by tommasini** [comment]
@jonybur can we include recordings on this video on what was tested?

File: app/components/UI/OptinMetrics/index.js [hunk 1]
```diff
@@ -299,8 +299,15 @@ class OptinMetrics extends PureComponent {
...
    */
   onCancel = async () => {
+    const {
+      isDataCollectionForMarketingEnabled,
+      setDataCollectionForMarketing,
+    } = this.props;
     setTimeout(async () => {
       const { clearOnboardingEvents, metrics } = this.props;
+      if (isDataCollectionForMarketingEnabled === null) {
+        setDataCollectionForMarketing(false);
```
Comment (NicolasMassart) [inline app/components/UI/OptinMetrics/index.js:309]: S: check for this function from prop to be defined before use. It's JS file, nothing prevents an undefined prop.

File: app/components/UI/OptinMetrics/index.js [hunk 2]
```diff
@@ -314,11 +321,20 @@ class OptinMetrics extends PureComponent {
...
+      metrics,
+      isDataCollectionForMarketingEnabled,
+      setDataCollectionForMarketing,
+    } = this.props;
     await metrics.enable();
     InteractionManager.runAfterInteractions(async () => {
       // add traits to user for identification
 
+      if (isDataCollectionForMarketingEnabled === null) {
+        setDataCollectionForMarketing(false);
```
Comment (NicolasMassart) [inline app/components/UI/OptinMetrics/index.js:335]: S: check for this function from prop to be defined before use. It's JS file, nothing prevents an undefined prop.

File: app/components/UI/OptinMetrics/index.js (hunk 1 above)
Comment (jonybur) [reply app/components/UI/OptinMetrics/index.js:309]: Done

File: app/components/UI/OptinMetrics/index.js (hunk 2 above)
Comment (jonybur) [reply app/components/UI/OptinMetrics/index.js:341]: Done

**Comment by hesterbruikman** [comment]
Recording of following test scenario:
- Fresh onboard
- Marketing consent opt-in checkbox is shown:

#### Deny marketing consent during onboarding
https://github.com/user-attachments/assets/d7cfe8ac-8038-4d4f-b0cb-23cd09587ab6


#### Opt in to marketing consent during onboarding
_Note that this video shows the checkbox as pre-checked. This is not the actual behavior; it's because I forgot to start recording and had already checked it_

https://github.com/user-attachments/assets/7819b5d2-c851-48d9-a1b2-d4e051e36325

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**